src/components/asset-list/RecyclerAssetList2/Claimable.tsx

```diff
@@ -1,17 +1,85 @@
-import React, { useMemo } from 'react';
+import React, { memo, useMemo } from 'react';
 import { Box, Inline, Stack, Text } from '@/design-system';
 import { FasterImageView } from '@candlefinance/faster-image';
 import { ButtonPressAnimation } from '@/components/animations';
 import { deviceUtils } from '@/utils';
 import Routes from '@/navigation/routesNames';
 import { ExtendedState } from './core/RawRecyclerList';
-import { convertAmountToNativeDisplayWorklet } from '@/helpers/utilities';
 import { analytics } from '@/analytics';
 import { ChainImage } from '@/components/coin-icon/ChainImage';
 import { ChainId } from '@/state/backendNetworks/types';
 import { Claimable as ClaimableType } from '@/resources/addys/claimables/types';
+import { DEVICE_WIDTH } from '@/utils/deviceUtils';
+import { getClaimableName } from '@/resources/addys/claimables/utils';
 
 const RAINBOW_ICON_URL = 'https://rainbowme-res.cloudinary.com/image/upload/v1694722625/dapps/rainbow-icon-large.png';
+const avgCharWidth = 7;
+const estimatedHorizontalPadding = 250;
+const maxChars = Math.floor((DEVICE_WIDTH - estimatedHorizontalPadding) / avgCharWidth);
+
+const NativeCurrencyDisplay = memo(function NativeCurrencyDisplay({ assets }: { assets: ClaimableType['assets'] }) {
```

Comment:
this could use some more testing. Seemed good from my limited testing, but might be brittle with super long token names